File: primatis-web/src/app/catalogue/models/author-response.ts
/**
 * Voir `be.primatis.catalogue.dto.AuthorResponse` côté backend. `fullName`
 * n'est pas une clé métier unique — deux Authors distincts peuvent porter le
 * même nom.
 */
export interface AuthorResponse {
  id: number;
  fullName: string;
  birthDate: string | null;
  deathDate: string | null;
  nationality: string | null;
  biography: string | null;
}

File: primatis-api/src/main/java/be/primatis/catalogue/dto/AuthorResponse.java
package be.primatis.catalogue.dto;

import be.primatis.catalogue.Author;

import java.time.LocalDate;
import java.util.Objects;

/**
 * Contrat REST de lecture d'un {@code Author} (DEV-06.3). Mapping 1:1, aucune
 * transformation ni déduction. {@code fullName} n'est pas une clé métier
 * unique (aucune contrainte {@code UNIQUE} en base) : deux Authors distincts
 * peuvent légitimement porter le même nom, ce DTO n'introduit donc aucune
 * notion d'unicité.
 */
public record AuthorResponse(
        Long id,
        String fullName,
        LocalDate birthDate,
        LocalDate deathDate,
        String nationality,
        String biography) {

    public static AuthorResponse from(Author author) {
        Objects.requireNonNull(author, "author");
        return new AuthorResponse(
                author.getId(),
                author.getFullName(),
                author.getBirthDate(),
                author.getDeathDate(),
                author.getNationality(),
                author.getBiography());
    }
}
